api/v1beta1/common.go

```diff
@@ -27,6 +27,91 @@ import (
 // NodeSection is a specification of the node attributes
 type NodeSection struct {
```

Comment:
There are a lot of fields reproduced between NodeSection and NodeTemplate. I think we need to find a way to have a common struct that can be reused. In the existing design, both Role and Node used the common NodeSection struct.

Reply:
Can we use a common struct for these fields somehow?

Reply:
Yeah, I'll have a better think about this. I agree. Initially, I was trying to nest NodeSection under itself which didn't work. ie, I was trying to do:
```
// NodeTemplate is a specification of the node attributes
type NodeTemplate struct {
	// +kubebuilder:validation:Required
	// Nodes - Map of Node Names and node specific data. Values here override defaults in the
	// upper level section.
	Nodes map[string]NodeTemplate `json:"nodes"`

```

Because we basically do need to repeat it to be able to override everything at the node level. For whatever reason, that failed and I moved to using `NodeSection` and `NodeTemplate` which were basically copies of one another.

I'll have a bit of a think about how we might be able to achieve this. Maybe we can have a quick brainstorm if I can't come up with any better plans.

Reply:
i need to read more of this pr but I'm a little confused by the breakdown of filed in these two types also 

ideally we never need to have the same filed in two places in the same CRD if it hosts the same info.


NetworkAttachments is oke to be in both node and node template if one is used to populate the other but we need to be carful not to allow conflict to happen or require people to keep both in sync.

validation or mutation webhooks can help but i think we should try and ensure the CRD leds its self to a single way to do thing in a natural way.

i need to properly load the code in my mind to comment though on how to change this as i don't fully understand the relationships currently.

i think in this case we only need NetworkAttachments at the NodeTemplate level in the spec since we do not expect different nodes in anodeset to have diffefnt networking. but i have not really thought the implications of that.

i.e. does it prevent a usecase we want to enable?